pkg/connections/stomp/connection_builder.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stomp
+
+import (
+	"github.com/container-mgmt/messaging-library/pkg/client"
+)
+
+// ConnectionBuilder is a helper struct for building connections
+type ConnectionBuilder struct {
+	BrokerHost   string
+	BrokerPort   int
+	UserName     string
+	UserPassword string
+	UseTLS       bool
+	InsecureTLS  bool
+}
+
+// Build a new connection object
+func (b *ConnectionBuilder) Build() (c client.Connection, err error) {
```

Comment:
I think the builder shouldn't have a Build() method and we should use:
``` go
func NewConnection(b *Connection) (*ClientConnection, error)

```

That way you can easily call it using:
``` go
c, err := NewConnection(&ConnectionBuilder{
		BrokerHost:   "myHost",
		BrokerPort:   123,
...
})

```
and not having to create a builder instance.
Like the example here:
http://www.theelements.org/golang/2016/05/30/builder-pattern-in-golang.html

Reply:
:+1: Done